mkiterian/bucket-list-app-api
# Ft register resource and endpoint 149365301

#### What does this PR do?
Add a register endpoint for the bucket list REST API
#### Description of Task to be completed?
Implement functionality for the endpoint:
- /api/v1/auth/register
A user should be able to register by supplying username, email, password and confirm the password
#### How should this be manually tested?
Sending a post request via the API with a username, email password and confirmation password either through postman or using curl in the terminal as shown: 
- curl -H "Content-Type: application/json" -X POST -d '{"username":"username", "email": "email", "password":"password", "confirm_password":"password"}' http://localhost:5000/api/v1/auth/register
should result in the user receiving a 'user successfully registered!' message
- If the username is missing in the request the user should receive a 'username is required' message
- If the email is missing in the request the user should receive an 'email is required' message
- If the password is missing or does not match the confirm password, the user should receive a 'password should match confirm password' message
#### What are the relevant pivotal tracker stories?
#149365301
